packages/dart_frog/lib/src/body_parsers/form_data.dart

```diff
@@ -37,3 +54,139 @@ bool _isFormUrlEncoded(ContentType? contentType) {
   if (contentType == null) return false;
   return contentType.mimeType == formUrlEncodedContentType.mimeType;
 }
+
+bool _isMultipartFormData(ContentType? contentType) {
+  if (contentType == null) return false;
+  return contentType.mimeType == multipartFormDataContentType.mimeType;
+}
+
+FormData _extractFormUrlEncodedFormData({required String body}) {
+  return FormData(Uri.splitQueryString(body), {});
+}
+
+Future<FormData> _extractMultipartFormData({
+  required Map<String, String> headers,
+  required Stream<List<int>> bytes,
+}) async {
+  final contentType = headers[HttpHeaders.contentTypeHeader]!;
+  final mediaType = MediaType.parse(contentType);
+  final boundary = mediaType.parameters['boundary'];
+  final transformer = MimeMultipartTransformer(boundary!);
+
+  final fields = <String, String>{};
+  final files = <String, UploadedFile>{};
+
+  await for (final part in transformer.bind(bytes)) {
+    final contentDisposition = part.headers['content-disposition'];
+    if (contentDisposition == null) continue;
+    if (!contentDisposition.startsWith('form-data;')) continue;
+
+    final values = RegExp('(?:(?<key>[a-zA-Z0-9-_]+)="(?<value>.*?)";*)+')
+        .allMatches(contentDisposition)
+        .fold(<String, String>{}, (map, match) {
+      return map..[match.namedGroup('key')!] = match.namedGroup('value')!;
+    });
+
+    final name = values['name']!;
+    final fileName = values['filename'];
+
+    if (fileName != null) {
+      files[name] = UploadedFile(
+        fileName,
+        ContentType.parse(part.headers['content-type'] ?? 'text/plain'),
+        part,
+      );
+    } else {
+      final bytes = (await part.toList()).fold(<int>[], (p, e) => p..addAll(e));
+      fields[name] = utf8.decode(bytes);
+    }
+  }
+
+  return FormData(fields, files);
+}
+
+/// {@template form_data}
+/// The fields and files of received form data request.
+/// {@endtemplate}
+class FormData with MapMixin<String, String> {
+  /// {@macro form_data}
+  const FormData(this._fields, this._files);
```

Comment:
Thoughts on making these named parameters for improved readability and flexibility?